protocol/strict-derivation.md

```diff
@@ -0,0 +1,269 @@
+timestamp, which is similar to applying Holocene rules retroactively.
+- Channel Bank:
+  1. All channels that don't consist of contiguous frames that include the first, are dropped. 
+  2. All but the youngest (in terms of L1 inclusion) channel are dropped.
+  3. This optional left-over channel
+     will be the start staging channel for the Holocene channel bank.
+
+The principle of eager derivation of later stages should guarantee at this point that all batches that could have
+been applied directly on top of the current safe chain have already been exhausted from the batch
+queue. So what is left should be `undecided` and `future` batches. (Note: I'm not sure on this one.)
+I see two options on how to handle a non-empty batch queue at this point:
+- Option 1: Drop future batches, continue resolving undecided batches, if any are left, and apply
+new Holocene rules.
+- Option 2: The Batch Queue will just start applying Holocene rules from this moment onwards. This will then
+automatically derive gaps as empty batches, discard out of order batches, etc. A problem with this
+option is that the sync start for a while after Holocene would still need to apply the pre-Holocene
+algorithm to restore the derivation pipeline state.
+
+I'm undecided at this point which option is better. I'd prefer to pick the one that leads to a
+simpler implementation, minimizing complexity where possible. Since almost always a healthy batcher
+implementation during normal network conditions will already produce batches according to Holocene
+rules, we don't need to optimize for the prettiest activation rules, as long as they work
+consistently.
+
+Another open question is how to handle span batches that come from pre-Holocene channels.
+I propose that, for simplicity of implementation, if they are found to be valid as a span batch, to
+just apply the new Partial Span Batch Validity rules even though those span batches were derived
+from pre-Holocene L1 blocks.
+
+# Alternatives Considered
+
+We could implement only some of the three changes to derivation. However, the synergies among the
+changes make it worthwile to apply them at once. E.g. strict batch ordering simplifies several
+stages in the derivation pipeline, thereby simplifying the reasoning about aspects like the right
+sync start behavior, which helps in implementing partial span batch validity more easily.
+Furthermore, the structural changes necessary in the derivation pipeline to apply multiple subsets
+of changes over multiple hardforks would create more work that can be saved if applying all those
+derivation changes in one go.
+
+# Risks & Uncertainties
+
+## Unsafe reorgs
+
+Before Steady Batch Derivation, invalid batches got chances to be replaced by valid future
+batches. Because they will now be immediately derived as deposit-only blocks, there is a theoretical
+heightened risk for unsafe head reorgs. However, we haven't experienced this on OP Mainnet or other mainnet
```

Comment:
Would a search of op-mainnet logs help identify if we've had situations which would cause Unsafe Reorgs?

Reply:
yes, but we retain logs for only 30 days unfortunately...